school_attendance/models/school_attendance.py

```diff
@@ -0,0 +1,336 @@
+        compute="_compute_total",
+        store=True,
+        help="Total Students in class",
+        string="Total Students",
+    )
+    total_presence = fields.Integer(
+        compute="_compute_present",
+        store=True,
+        string="Present Students",
+        help="Present Student",
+    )
+    total_absent = fields.Integer(
+        compute="_compute_absent",
+        store=True,
+        string="Absent Students",
+        help="Absent Students",
+    )
+
+    _sql_constraints = [
+        (
+            "attend_unique",
+            "unique(standard_id,user_id,date)",
+            "Attendance should be unique!",
+        )
+    ]
+
+    def _open_student_attendance_sheet(self, view_xmlid):
+        """Method to open the wizard to fill
+        the attendance of the students"""
+        wiz = self.env["attendance.sheet.wiz"].create({})
+        view_id = self.env.ref("school_attendance.%s" % view_xmlid).id
+        return {
+            "name": "Student Attendance Sheet",
+            "type": "ir.actions.act_window",
+            "view_type": "form",
+            "view_mode": "form",
+            "res_model": "attendance.sheet.wiz",
+            "target": "new",
+            "res_id": wiz.id,
+            "view_id": view_id,
+            "context": self.env.context,
+        }
+
+    def open_student_attendance_sheet(self):
+        return self._open_student_attendance_sheet(
+            "student_attendance_sheet_wiz"
+        )
+
+    @api.onchange("user_id")
+    def onchange_check_faculty_value(self):
+        """Onchange method for faculty"""
+        if self.user_id:
+            self.standard_id = False
+
+    @api.constrains("date")
+    def validate_date(self):
+        """Constraint to check selected attendance date"""
+        if self.date > fields.Date.today():
+            raise ValidationError(
+                _("Date should be less than or equal to current date!")
+            )
+
+    def attendance_draft(self):
+        """Change the state of attendance to draft"""
+        self.state = "draft"
+
+    def attendance_validate(self):
+        """Method to validate attendance."""
+        acadmic_month_obj = self.env["academic.month"]
+        attendance_sheet_obj = self.env["attendance.sheet"]
+
+        for line in self:
+            month_ids = acadmic_month_obj.search(
+                [
+                    ("date_start", "<=", line.date),
+                    ("date_stop", ">=", line.date),
+                    ("year_id.date_start", "<=", line.date),
+                    ("year_id.date_stop", ">=", line.date),
+                ],
+                limit=1,
+            )
+            att_sheet_ids = attendance_sheet_obj.search(
+                [
+                    ("month_id", "in", month_ids.ids),
+                    ("year_id", "=", month_ids.year_id.id),
+                ]
+            )
+            attendance_sheet_id = att_sheet_ids and att_sheet_ids[0] or False
```

Comment:
Use limit=1 to search att_sheet_ids will helps to improve this line.
OR
no need this line, directly check if condition with att_sheet_ids.